modules/db/src/main/scala/gem/dao/TargetDao.scala

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package dao
+
+import doobie._
+import doobie.implicits._
+import gem.dao.meta._
+import gem.dao.composite._
+import gem.enum.TrackType
+import shapeless._
+
+object TargetDao {
+  import EnumeratedMeta._
+  import EphemerisKeyComposite._
+  import ProperMotionComposite._
+  import TaggedCoproduct._
+  import Track._
+
+  def select(id: Int): ConnectionIO[Option[Target]] =
+    Statements.select(id).option
+
+  def insert(target: Target): ConnectionIO[Int] =
+    Statements.insert(target).withUniqueGeneratedKeys[Int]("id")
+
+  def update(id: Int, target: Target): ConnectionIO[Int] =
+    Statements.update(id, target).run
+
+  def delete(id: Int): ConnectionIO[Int] =
+    Statements.delete(id).run
+
+  object Statements {
+
+    // Track is laid out as a tagged coproduct: (tag, sidereal, nonsidereal).
+    private implicit val TaggedTrackComposite: Composite[Track] = {
+
+      // We map only the ephemeris key portion of the nonsidereal target here, and we only need to
+      // consider the Option[Nonsidereal] case because this is what the coproduct encoding needs.
+      implicit val compositeOptionNonsidereal: Composite[Option[Nonsidereal]] =
+        Composite[Option[EphemerisKey]].imap(_.map(Nonsidereal.empty))(_.map(_.ephemerisKey))
+
+      // Construct an encoder for track constructors, tagged by TrackType.
+      val enc = Tag[Sidereal](TrackType.Sidereal)       :+:
+                Tag[Nonsidereal](TrackType.Nonsidereal) :+: TNil
+
```

Comment:
Without the change to enum above the tag type here is inferred as `TrackType with Product with Serializable` which doesn't have a `Meta` instance. With the change it's just `TrackType` which is what we want.

Reply:
The implementation of `Tag` and `TNil` is below in `TaggedCoproduct.scala`.